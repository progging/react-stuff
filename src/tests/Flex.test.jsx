import React from "react"
import {Flex} from ".."
import {mount} from 'enzyme'
import {mountToJson} from 'enzyme-to-json'
import 'jest-styled-components'

test("Component should have Flexbox properties", () => {

    const component = mount(
        <Flex alignItems="center" justifyContent="space-between">
            <div/>
        </Flex>
    )

    expect(mountToJson(component)).toMatchSnapshot()
    expect(component).toHaveStyleRule('display', 'flex')
    expect(component).toHaveStyleRule('align-items', 'center')
    expect(component).toHaveStyleRule('justify-content', 'space-between')
})

test("Inline flex switch should be reflected in `display` property", () => {

    const component = mount(
        <Flex inlineFlex={true}>
            <div/>
        </Flex>
    )

    expect(mountToJson(component)).toMatchSnapshot()
    expect(component).toHaveStyleRule('display', 'inline-flex')
})