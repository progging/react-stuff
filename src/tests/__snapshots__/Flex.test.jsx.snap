// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component should have Flexbox properties 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

<Flex
  alignItems="center"
  justifyContent="space-between"
>
  <styled.div
    css={
      Array [
        Object {
          "cssProp": "align-items",
          "val": "center",
        },
        Object {
          "cssProp": "justify-content",
          "val": "space-between",
        },
      ]
    }
  >
    <div
      className="c0"
    >
      <div />
    </div>
  </styled.div>
</Flex>
`;

exports[`Inline flex switch should be reflected in \`display\` property 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

<Flex
  inlineFlex={true}
>
  <styled.div
    css={Array []}
    inlineFlex={true}
  >
    <div
      className="c0"
    >
      <div />
    </div>
  </styled.div>
</Flex>
`;
